### PR TITLE
Fix declarations of functions

### PR DESCRIPTION
**Pull request by janisozaur**
* Add `static` where needed
* Explicitly name inheritance type
* Add missing include
* Drop `#pragma once` from .cpp files

### CODE REVIEW

File: src/CalcManager/CEngine/scicomm.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ using namespace CalcEngine;
 //
 //  returns a virtual number for precedence for the operator. We expect binary operator only, otherwise the lowest number
 // 0 is returned. Higher the number, higher the precedence of the operator.
-INT NPrecedenceOfOp(int nopCode)
+static INT NPrecedenceOfOp(int nopCode)
```
Comment (danbelcher-MSFT) [inline src/CalcManager/CEngine/scicomm.cpp]: I think the better fix here is to move these functions to an anonymous namespace.  Same for the other instances below.

**Comment by janisozaur** [comment]
Updated

File: src/CalcManager/CalcException.h [hunk 2]
```diff
@@ -15,12 +15,3 @@ class CalcException : std::exception
 private:
     HRESULT m_hr;
 };
-
-void IFT(HRESULT hr)
```
Comment (janisozaur) [inline src/CalcManager/CalcException.h]: This was moved into `History.cpp`, the only consumer of `CalcException` is also in there, should I move it into `History.cpp` as well and remove this file?

Comment (danbelcher-MSFT) [reply src/CalcManager/CalcException.h]: Yep, I think that's a good change.  Let's limit visibility of functions to places where they are needed.

Comment (janisozaur) [reply src/CalcManager/CalcException.h]: Updated

File: src/CalcManager/CEngine/scicomm.cpp [hunk 3]
```diff
@@ -30,24 +30,26 @@ using namespace CalcEngine;
 //
 //  returns a virtual number for precedence for the operator. We expect binary operator only, otherwise the lowest number
 // 0 is returned. Higher the number, higher the precedence of the operator.
```
Comment (danbelcher-MSFT) [inline src/CalcManager/CEngine/scicomm.cpp]: Move this comment to be with the `NPrecedenceOfOp` function.

File: src/CalcManager/CEngine/History.cpp [hunk 4]
```diff
@@ -2,18 +2,41 @@
...
 #include "CalculatorVector.h"
 #include "ExpressionCommand.h"
-#include "CalcException.h"
 
 constexpr int ASCII_0 = 48;
 
 using namespace std;
 using namespace CalcEngine;
 
+class CalcException : public std::exception
```
Comment (danbelcher-MSFT) [inline src/CalcManager/CEngine/History.cpp]: On second look, this class isn't needed.  It doesn't override `std::exception::what` and nowhere in the codebase explicitly catches a `CalcException`.

File: src/CalcManager/CEngine/scicomm.cpp (hunk 3 above)
Comment (janisozaur) [reply src/CalcManager/CEngine/scicomm.cpp]: done

File: src/CalcManager/CEngine/History.cpp (hunk 4 above)
Comment (janisozaur) [reply src/CalcManager/CEngine/History.cpp]: done

**Comment by danbelcher-MSFT** [comment]
Hey @janisozaur, looks like there's merge conflicts now.  If you fix them up we can get the change merged.  Thanks!

**Comment by janisozaur** [comment]
I'll try updating this in the evening, some 10h from now.

**Comment by janisozaur** [comment]
Updated

**Pull request merged**